pkg/autoops/api/progressive_rollout.go

```diff
@@ -475,7 +478,13 @@ func (s *AutoOpsService) ExecuteProgressiveRollout(
 		); err != nil {
 			return err
 		}
-		if err := ftStorage.UpdateFeature(ctx, feature, req.EnvironmentNamespace); err != nil {
+		if _, err := s.featureClient.UpdateFeature(ctx, &featureproto.UpdateFeatureRequest{
```

Comment:
Calling an API inside of a transaction MySQL is not a good idea.
If we need to increment the version, we should implement it here.
This could easily get into the wrong state where the API succeeds, but MySQL doesn't.

Reply:
I agree with that. It is not a good idea to call API inside the transaction. But I don't think updating the feature directly and create a domain event here is a good way. So I will change this to calling UpdateFeature outside transaction (the way in which the autoops does).

Reply:
If the auto ops service changes the feature entity, I don't see why we can't create the event to publish it here.
The issue remains even after moving the API calls outside of the transaction. The Progressive Rollout will be updated as triggered, and if the API call fails, the Flag won't change its state, causing bugs and making it harder to debug.

To avoid other issues, it will be safer to update the flag in the same transaction and create the domain event for it here.